ducktape/tests/loader.py

```diff
@@ -138,11 +152,20 @@ def discover(self, test_discovery_symbols):
             if len(test_info_list) == 0:
                 raise LoaderException("Didn't find any tests for symbol %s." % symbol)
 
+        # Assemble the test context objects
+        test_context_list = []
+        for t in test_info_list:
+            if t.regression_variable_selector is None:
+                test_context_list.append(
+                    TestContext(self.session_context, t.module_name, t.cls, t.function, t.injected_args))
+            else:
+                target_test_context = TestContext(self.session_context, t.module_name, t.cls, t.function, t.injected_args)
```

Comment:
The `TestContext` being constructed in both branches of this conditional are the same and can be lifted outside it. The only difference is whether you wrap it in the `DefaultRegressionTest.create_test_context`.